server/webapp/WEB-INF/rails.new/app/controllers/stages_controller.rb

```diff
@@ -245,8 +245,9 @@ def convert_to_scale duration_array
     stage_duration_array
   end
 
-  def has_permission_to_view_settings
+  def can_view_settings?
```

Comment:
Make this a helper method using `helper_method :can_view_settings?` and access it directly from the view, instead of via `@can_user_view_settings`

If performance is a concern, you can always memoize the result —

```ruby
def can_view_settings?
  @can_view_settings ||= begin
    group_name = ...
    !group_name.blank? && ...
  end
end
```